indigo_app/views/tasks.py

```diff
@@ -259,31 +259,24 @@ def post(self, request, *args, **kwargs):
                 if not has_transition_perm(state_change, self):
                     raise PermissionDenied
                 state_change(user)
-                if verb == 'submitted':
+
+                # assignee side effects of different state changes
+                if change == 'submit':
                     task.last_assigned_to = task.assigned_to
                     task.assigned_to = None
-                    action.send(user, verb=verb, action_object=task,
-                                place_code=task.place.place_code)
-                    messages.success(request, u"Task '%s' has been submitted for review" % task.title)
-                elif verb == 'unsubmitted' and task.last_assigned_to:
-                    assignee = task.last_assigned_to
-                    task.assigned_to = assignee
-                    if user.id == assignee.id:
-                        action.send(user, verb='unsubmitted and picked up', action_object=task,
-                                    place_code=task.place.place_code)
-                        messages.success(request, u"You have unsubmitted and picked up the task '%s'" % task.title)
-                    else:
-                        action.send(user, verb='unsubmitted and reassigned', action_object=task,
-                                    target=assignee,
-                                    place_code=task.place.place_code)
-                        messages.success(request, u"Task '%s' has been unsubmitted and reassigned" % task.title)
-
-                else:
-                    if verb == 'closed' or verb == 'cancelled':
-                        task.assigned_to = None
-                    action.send(user, verb=verb, action_object=task,
-                                place_code=task.place.place_code)
-                    messages.success(request, u"Task '%s' has been %s" % (task.title, verb))
+                if change == 'unsubmit' and task.last_assigned_to:
```

Comment:
Are you happy that the special case messaging is dropped?

Reply:
yeah I think it makes more sense as is with the new wording – 'requested changes' implies reassigning?

Reply:
or are you talking about when the original assignee unsubmits the task? 
that's probably less likely now that the button reads 'request changes'?

Reply:
Yes, I was just confirming you've thought through this, because I haven't :)